.github/workflows/ci.yml

```diff
@@ -11,6 +11,7 @@ on:
 
 jobs:
   ci:
+    if: github.repository == 'rust-lang/rustc-dev-guide'
```

Comment:
The whole issue about this run on forks confuses me. Actions shouldn't run on forks by default...

Reply:
This still avoids running workflows by mistakes.

Reply:
How would you run it by mistake? You'd have to modify the workflow file, right? If you're intentionally changing CI I don't think we should make it harder.

Reply:
> Actions shouldn't run on forks by default...

Yeah, I feel confused about that. Is there a setting somewhere in the config that changed the behavior?

Reply:
> You'd have to modify the workflow file, right? If you're intentionally changing CI I don't think we should make it harder.

No, without this, workflows can be triggered even if you don't edit the config file. If you pull upstream commits and push to fork's master, for example, the workflows will run on your fork.

I just forked rust-lang/docs.rs and the actions' config looks like:
![image](https://user-images.githubusercontent.com/25030997/111885512-9db39f00-8a0b-11eb-9b50-6890134eb698.png)

Actually, workflows are currently triggered on @camelid's and my fork repos, that's why I made this change.

Reply:
> I just forked rust-lang/docs.rs and the actions' config looks like:
> ![image](https://user-images.githubusercontent.com/25030997/111885512-9db39f00-8a0b-11eb-9b50-6890134eb698.png)

But my fork of docs.rs (or any other repo) never runs GHA...

Reply:
Ah, sorry, it was confusing. You have to click this to enable it actually:
![image](https://user-images.githubusercontent.com/25030997/111885721-e3249c00-8a0c-11eb-8c91-0959c3b916ec.png)

I just mean it _can be_ triggered without editing workflows/*.yml

Reply:
I'm really confused though - if someone explicitly enables ci on their fork, why wouldn't we want it to run?

Reply:
Hmm, maybe the issue is that workflows were added *after* we made our forks, where as for other repos they were added before we forked?

Reply:
I don't understand though - what here is a bug? Everything @JohnTitor's shown looks to me as if it's working as expected.

Reply:
The issue is that by default the rustc-dev-guide GHA runs on forks when you push to your fork's master. No other repo I know of does that.

Reply:
> Hmm, maybe the issue is that workflows were added after we made our forks, where as for other repos they were added before we forked?

Oh I didn't know that behavior, hmm, then it should be more helpful than I thought. So this would be a problem for all the contributors who forked this repo before we introduced GHA if it's true, and it'll cause the notifications and eat the concurrent job limits.

Anyway, I don't see any negative effects by this so I'd like to add this stopper for further issues.

Reply:
Interestingly, this just happened to me on my fork of the Rust playground. It hasn't happened before. I wonder if this is a new GitHub behavior?

Reply:
So, does someone still have a concern about this? I think this is the only blocker here but I don't see any disadvantage.

Reply:
I'm still a little confused why this is necessary, but I don't push to master branches on forks, so if you think it's useful I'm fine merging it.